tokio/src/runtime/driver/op.rs

```diff
@@ -0,0 +1,161 @@
+use crate::runtime::Handle;
+use io_uring::cqueue;
+use io_uring::squeue::Entry;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use std::task::Waker;
+use std::{io, mem};
+
+#[derive(Debug)]
+pub(crate) enum Lifecycle {
+    /// The operation has been submitted to uring and is currently in-flight
+    Submitted,
+
+    /// The submitter is waiting for the completion of the operation
+    Waiting(Waker),
+
+    /// The submitter no longer has interest in the operation result. The state
+    /// must be passed to the driver and held until the operation completes.
+    Cancelled(#[allow(unused)] Box<dyn std::any::Any + Send + 'static>),
```

Comment:
I'm fine merging this PR w/ the box here, but I am curious why it is needed.

Reply:
Removed the Box from the Cancelled variant, and introduced an enum type that holds the operation data.
[cdb176e](https://github.com/tokio-rs/tokio/pull/7320/commits/cdb176e54594d957fcdbc15297e75fe929d970d2)